cql3/expr/expression.hh

```diff
@@ -106,24 +155,50 @@ enum class comparison_order : char {
 
 /// Operator restriction: LHS op RHS.
 struct binary_operator {
-    std::unique_ptr<expression> lhs;
+    nested_expression lhs;
     oper_t op;
     ::shared_ptr<term> rhs;
     comparison_order order;
 
     binary_operator(expression lhs, oper_t op, ::shared_ptr<term> rhs, comparison_order order = comparison_order::cql);
-
-    binary_operator(const binary_operator& x);
-    binary_operator& operator=(const binary_operator&);
-    binary_operator(binary_operator&& x) = default;
-    binary_operator& operator=(binary_operator&& x) = default;
 };
 
 /// A conjunction of restrictions.
 struct conjunction {
     std::vector<expression> children;
 };
 
+// Gets resolved eventually into a column_value.
+struct unresolved_identifier {
+    shared_ptr<column_identifier_raw> ident;
+
+    ~unresolved_identifier();
+};
+
+// An attribute attached to a column mutation: writetime or ttl
+struct column_mutation_attribute {
+    enum class attribute_kind { writetime, ttl };
+
+    attribute_kind kind;
+    // note: unresolved_identifier is legal here now, later column_value.
```

Comment:
A bit of both. It's legal today, before prepare. Tomorrow, we'll implement preparing unresolved_identifier into an column_value and it will be legal after prepare().

Reply:
I like this explanation and wish I read it in the comment. :)

Reply:
added